include/wayland/mir/wayland/resource.h

```diff
@@ -26,19 +26,25 @@ namespace mir
 {
 namespace wayland
 {
+class Client;
+
 class Resource
     : public virtual LifetimeTracker
 {
+private:
+    std::shared_ptr<Client> owned_client;
+
 public:
     template<int V>
     struct Version
     {
     };
 
     Resource(wl_resource* resource);
+    virtual ~Resource();
 
     wl_resource* const resource;
-    wl_client* const client;
+    Client& client;
```

Comment:
I'm not entirely convinced that making `client` look like a member variable is the best (I'd be happier with a pointer). But that's a point we often disagree on and not worth blocking over

Reply:
I also note that almost all the use of client is with `client.raw_client()`: maybe that could be encapsulated as a helper function?

Reply:
I tend to agree about refs/ptrs, changing.

Another option instead of `raw_client()` is to make a `wl_client*` operator on `mw::Client`, but we can put this off to a future PR.